tests/slicing/sources/bitcast2.c

```diff
@@ -1,15 +1,15 @@
-_Static_assert(sizeof(int) == 4, "This test assumes sizeof(int) == 4");
+#include <limits.h>
 
 /* test accessing bytes in int */
 int main(void) {
-    int a;
-    a = 0;
+    int a = 0;
     char *byte = (char *) &a;
-    byte[0] = 0xab;
-    byte[1] = 0xab;
-    byte[2] = 0xab;
-    byte[3] = 0xab;
-    test_assert(a == 0xabababab);
+
+    int result = 0;
+    for (unsigned i = 0; i < sizeof(int); i++)
+        result |= (byte[i] = 0x1b) << i * CHAR_BIT;
```

Comment:
This completely changes the test, at least from the point of view of the slicer. Accessing the bytes via constants (e.g., `byte[1]`) leads to a precise knowledge of what memory is accessed, while accessing it via a variable (`bytes[i]`) leads to over-approximating the precise knowledge. Please, keep the spirit of the test as it was (accessing bytes via constants). If you want this test with the loop, just add it as a new test (but check that there is not a similar test already).

Reply:
Oh, I see. I wanted to make the test platform as platform independent as possible so didn't even think about this. I'll revert this change and add this test as a new one. There is already a rather similar one but that one is accessing the bytes by `*byte; byte++` pattern and not by `byte[i]` so I guess that's ok.

Reply:
Nevertheless, I'll change the the type of the `char` pointer to be explicitly `unsigned` to silence a `clang-tidy` warning about implementation defined behaviour of a narrowing conversion from `int` to `char` for values greater than `127` which `0xab` is.